src/options/parser.rs

```diff
@@ -1,748 +1,217 @@
-
-    /// Whether to check for duplicate or redundant arguments.
-    strictness: Strictness,
-}
-
-impl<'a> MatchedFlags<'a> {
-
-    /// Whether the given argument was specified.
-    /// Returns `true` if it was, `false` if it wasn’t, and an error in
-    /// strict mode if it was specified more than once.
-    pub fn has(&self, arg: &'static Arg) -> Result<bool, OptionsError> {
-        self.has_where(|flag| flag.matches(arg))
-            .map(|flag| flag.is_some())
-    }
-
-    /// Returns the first found argument that satisfies the predicate, or
-    /// nothing if none is found, or an error in strict mode if multiple
-    /// argument satisfy the predicate.
-    ///
-    /// You’ll have to test the resulting flag to see which argument it was.
-    pub fn has_where<P>(&self, predicate: P) -> Result<Option<&Flag>, OptionsError>
-    where P: Fn(&Flag) -> bool {
-        if self.is_strict() {
-            let all = self.flags.iter()
-                          .filter(|tuple| tuple.1.is_none() && predicate(&tuple.0))
-                          .collect::<Vec<_>>();
-
-            if all.len() < 2 { Ok(all.first().map(|t| &t.0)) }
-                        else { Err(OptionsError::Duplicate(all[0].0, all[1].0)) }
-        }
-        else {
-            Ok(self.has_where_any(predicate))
-        }
-    }
-
-    /// Returns the first found argument that satisfies the predicate, or
-    /// nothing if none is found, with strict mode having no effect.
+pub use clap::Parser;
+use std::ffi::OsString;
+
+#[derive(Parser, Default)]
+#[command(author, version, about, long_about = None)] // Read from `Cargo.toml`
+#[clap(disable_help_flag = true)]
+pub struct Opts {
+    pub paths: Vec<OsString>,
+    /// Show hidden files.
+    #[arg(short, long, action = clap::ArgAction::Count)]
+    pub all: u8,
+    /// display extended file metadata as a table.
+    #[arg(short, long, action = clap::ArgAction::Count)]
+    pub long: u8,
+    /// list each file's Git status, if tracked or ignored.
+    #[arg(long, action = clap::ArgAction::Count)]
+    pub git: u8,
+    /// Display one entry per line.
+    #[arg(short = '1', long, action = clap::ArgAction::Count)]
+    pub oneline: u8,
+    ///recurse into directories as a tree.
+    #[arg(short = 'T', long, action = clap::ArgAction::Count)]
+    pub tree: u8,
+    /// display entries as a grid (default).
+    #[arg(short = 'G', long, action = clap::ArgAction::Count)]
+    pub grid: u8,
+    /// sort the grid across, rather than downwards.
+    #[arg(short = 'x', long, action = clap::ArgAction::Count)]
+    pub across: u8,
+    /// recurse into directories.
+    #[arg(short = 'R', long, action = clap::ArgAction::Count)]
+    pub recurse: u8,
+    /// display type indicator by file names.
+    #[arg(short = 'F', long, action = clap::ArgAction::Count)]
+    pub classify: u8,
+    /// 
+    #[arg(short = 'X', long, action = clap::ArgAction::Count)]
+    pub dereference: u8,
```

Comment:
Missing documentation here, and two other places below.

Also, I think the name should slightly more informative, like `dereference_links`.

Reply:
there wasnt any for those in the old help message

Reply:
Ah, I just realized that the variable name was probably linked to the actual flag on the command line.